### ISSUE TITLE
Cannot generate embedded PDB for Windows using cl.exe.

### ISSUE THREAD

**Issue by jfalcon** [curl/curl#7502]
### I did this

Building `libcurl` on a Windows box using the `winbuild` folder and Visual Studio 2022 Preview with the native x64 Developer Command prompt (_I was intentionally verbose with some of the arguments_):

```
nmake /nologo /f Makefile.vc mode=static vc=17 debug=yes GEN_PDB=yes CFLAGS_PDB=/Z7 RTLIBCFG=dll
  ENABLE_IDN=yes ENABLE_IPV6=yes ENABLE_SCHANNEL=yes ENABLE_SSPI=yes ENABLE_UNICODE=yes
  NGHTTP2_PATH=C:\Path\To\Nghttp2\install NGHTTP2_INC_DIR=C:\Path\To\Nghttp2\include
  SSH2_PATH=C:\Path\To\libssh2\install SSH2_INC_DIR=C:\Path\To\libssh2\include
  USE_IDN=true USE_IPV6=true USE_NGHTTP2=true USE_SCHANNEL=true USE_SSH2=true
  USE_SSPI=true USE_UNICODE=true USE_WINDOWS_SSPI=true USE_ZLIB=true
  WITH_DEVEL=C:\Path\To\Built\Libs WITH_PREFIX="C:\Path\To\Built\Libs
  WITH_NGHTTP2=static WITH_SSH2=static WITH_ZLIB=static
  ZLIB_PATH=%ZLIB_INSTALL_FOLDER% ZLIB_INC_DIR=%ZLIB_INSTALL_FOLDER%\include
```

### I expected the following

`libcurl` to be built out with the PDB file embedded into the library. But no matter what I do, I cannot get this to happen without directly editing `MakefileBuild.vc`, even tried manually overriding the `CFLAGS_PDB` macro as shown above to no avail. If you look in `MakefileBuild.vc` on line `65`, that overrides the `/Z7` flag set on line `52` since both arguments will end up on the cl.exe command line.

Which will produce the following warning since the `/Zi` appears _after_ the `/Z7` from line `52`:

```cl : Command line warning D9025 : overriding '/Z7' with '/Zi'```

Which means the PDB file is never embedded into the library.

### curl/libcurl version

7.78.0

### operating system

Windows 10 64-bit, Version 21H1 (OS Build 19043.1110)

**Comment by jfalcon** [comment]
And I realize I'm setting the `GEN_PDB=yes` flag, but if you look at the readme it says:

```GEN_PDB=<yes/no>`            - Generate Program Database (debug symbols for release build)```

Which at the very least should say `Generate External Program Database` since that's what it does and prevents you from generating one inline.

**Comment by jay** [comment]
> Which at the very least should say `Generate External Program Database`

Ok. In your case you may be able to use CC variable, and skip GEN_PDB.

**Comment by jfalcon** [comment]
Thanks for the reply. That's exactly what I did to keep the juju going. The makefile defaults to `/Z7` without `GEN_PDB`, so it works without it. Just wanted to mention it is all.

**Comment by bagder** [comment]
I don't understand. Does it mean that this issue is sorted or is there a change still needed/desired?

**Comment by jay** [comment]
> I don't understand. Does it mean that this issue is sorted or is there a change still needed/desired?

The first issue is that the GEN_PDB option to generate debug symbols for a release build does not mention that it is generated separately, so I've taken the reporter's suggestion and landed 86c9146.

The second issue is that we don't have an option to generate debug symbols for a release build that are included in the binaries. Frankly, I don't think this is necessary unless we see demand for it.

I've removed help wanted and am closing as not-a-bug.
